### PR TITLE
Ensure bindgen on android uses the correct compiler [ci full]

### PR DESCRIPTION
**Pull request by thomcc**
This took a detour, but now we don't have to do hacks to configure the build path. Without this, it doesn't build for me, but more importantly it's *really dodgy* to use different compilers here... Like, Oof, that's worrying. This needs to be fixed in rust-android-gradle as well, but for now this is good.

### CODE REVIEW

File: components/support/rc_crypto/nss/nss_sys/build.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ use std::{
     fs,
     path::{Path, PathBuf},
     process::Command,
+    // sync::atomic::{AtomicBool, Ordering},
```
Comment (eoger) [inline components/support/rc_crypto/nss/nss_sys/build.rs]: Nit: leftover comment.

File: components/support/rc_crypto/nss/nss_sys/build.rs [hunk 2]
```diff
@@ -56,7 +57,53 @@ fn env_str(name: &str) -> Option<String> {
     env::var(name).ok()
 }
 
+fn env_flag(name: &str) -> bool {
+    match env_str(name).as_ref().map(String::as_ref) {
+        Some("1") => true,
+        Some("0") => false,
+        Some(s) => {
+            println!(
```
Comment (eoger) [inline components/support/rc_crypto/nss/nss_sys/build.rs]: Will this prove to be annoying?

File: components/support/rc_crypto/nss/nss_sys/build.rs [hunk 3]
```diff
@@ -56,7 +57,53 @@ fn env_str(name: &str) -> Option<String> {
...
+            // different too, who knows.
+            target = "armv7a-linux-androideabi".to_owned();
+        }
+        for &path in &["toolchains", "llvm", "prebuilt", android_host_tag(), "bin"] {
+            buf.push(path);
+        }
+        buf.push(format!("{}{}-clang", target, ndk_api));
+        env::set_var("CLANG_PATH", buf);
+        // If we configured the right path, we don't need to separately set the includes.
+        // ANDROID_INCLUDES_NEED_CONFIGURING.store(false, Ordering::SeqCst);
```
Comment (eoger) [inline components/support/rc_crypto/nss/nss_sys/build.rs]: Leftover comment.

File: components/support/rc_crypto/nss/nss_sys/build.rs [hunk 4]
```diff
@@ -56,7 +57,53 @@ fn env_str(name: &str) -> Option<String> {
...
+            println!("cargo:warning=ANDROID_NDK_API_VERSION is unset. Trying unprefixed");
+        }
+        let mut target = env::var("TARGET").unwrap();
+        if target == "armv7-linux-androideabi" {
+            // Yeah, dunno why these are different. The binutils triple is
+            // different too, who knows.
+            target = "armv7a-linux-androideabi".to_owned();
+        }
+        for &path in &["toolchains", "llvm", "prebuilt", android_host_tag(), "bin"] {
+            buf.push(path);
```
Comment (eoger) [inline components/support/rc_crypto/nss/nss_sys/build.rs]: huh, maybe one day there will be a `&[&str]` `push`

File: components/support/rc_crypto/nss/nss_sys/build.rs [hunk 5]
```diff
@@ -56,7 +57,53 @@ fn env_str(name: &str) -> Option<String> {
...
+        let mut buf = PathBuf::from(env("ANDROID_NDK_ROOT").unwrap());
+        let ndk_api = env_str("ANDROID_NDK_API_VERSION")
+            .unwrap_or(DEFAULT_ANDROID_NDK_API_VERSION.to_owned());
+
+        if ndk_api.is_empty() {
+            println!("cargo:warning=ANDROID_NDK_API_VERSION is unset. Trying unprefixed");
+        }
+        let mut target = env::var("TARGET").unwrap();
+        if target == "armv7-linux-androideabi" {
+            // Yeah, dunno why these are different. The binutils triple is
```
Comment (eoger) [inline components/support/rc_crypto/nss/nss_sys/build.rs]: If you want to link the reference (https://developer.android.com/ndk/guides/other_build_systems):

(search for "For 32-bit ARM, the compiler is prefixed with armv7a-linux-androideabi")

File: components/support/rc_crypto/nss/nss_sys/build.rs [hunk 6]
```diff
@@ -56,7 +57,53 @@ fn env_str(name: &str) -> Option<String> {
     env::var(name).ok()
 }
 
+fn env_flag(name: &str) -> bool {
```
Comment (eoger) [inline components/support/rc_crypto/nss/nss_sys/build.rs]: Nice cleanup!

File: components/support/rc_crypto/nss/nss_sys/build.rs (hunk 2 above)
Comment (thomcc) [reply components/support/rc_crypto/nss/nss_sys/build.rs]: Doubt it. If it's a flag what else would you set it to?

**Pull request merged**